pylearn2/datasets/dataset.py

```diff
@@ -18,7 +18,7 @@ def __iter__(self):
         return self.iterator()
 
```

Comment:
get_batch_topo and get_batch_design are useful for quick and dirty experimentation.

While it is important for Pylearn2 to be flexible, it's also important for it to be easy to use. If I want to write a script that just loads a design matrix and runs it through an autoencoder's reconstruction function or something like that, I don't want to spend two hours reading the documentation about spaces and sources.

Rather than getting rid of get_batch_design and get_batch_topo, I'd suggest making them methods only of DenseDesignMatrix.